src/mame/jaleco/megasys1.cpp

```diff
@@ -365,6 +389,38 @@ void megasys1_state::megasys1B_monkelf_map(address_map &map)
 #define INTERRUPT_NUM_C INTERRUPT_NUM_B
 #define interrupt_C     interrupt_B
 
+void megasys1_state::megasys1c_handle_scanline_irq(int scanline)
```

Comment:
@angelosa SystemC IRQ timings were updated to match what's measured from actual pcb.
Probably St Dragon issue is related to the same thing, but unfortunately I don't have any systemA board to take reading from.